kernel/src/main/scala/cats/kernel/std/array.scala

```diff
@@ -0,0 +1,73 @@
+package cats.kernel
+package std
+
+import scala.{ specialized => sp }
+
+package object array extends ArrayInstances
```

Comment:
I'm fine with leaving the array instances in Algebra if people would prefer that.